MapboxNavigation/NavigationViewController.swift

```diff
@@ -608,16 +608,8 @@ extension NavigationViewController: TunnelIntersectionManagerDelegate {
 
 extension NavigationViewController: StyleManagerDelegate {
     
-    public func locationFor(styleManager: StyleManager) -> CLLocation {
-        guard let location = routeController.location else {
-            if let coordinate = routeController.routeProgress.route.coordinates?.first {
-                return CLLocation(latitude: coordinate.latitude, longitude: coordinate.longitude)
-            } else {
-                return CLLocation()
-            }
-        }
-        
-        return location
+    public func locationFor(styleManager: StyleManager) -> CLLocation? {
+        return routeController.location
```

Comment:
`return routeController.location ?? routeController.routeProgress.route.coordinates?.first` to make sure we have a location when we first apply day/night style? Should be safe to assume that the first coordinate on the route is in the same timezone as the user.